src/main/java/org/terasology/launcher/game/GameDownloader.java

```diff
@@ -57,7 +57,18 @@ public GameDownloader(File downloadDirectory, File tempDirectory, boolean saveDo
         if (downloadZipFile.exists() && (!downloadZipFile.isFile() || !downloadZipFile.delete())) {
             throw new IOException("Could not delete file! " + downloadZipFile);
         }
-        downloadURL = DownloadUtils.createFileDownloadUrlJenkins(jobName, buildNumber, DownloadUtils.FILE_TERASOLOGY_GAME_ZIP);
+
+        // If we have a matching Omega distribution for this game version then fetch that zip file instead
+        if (gameVersion.getOmegaNumber() != null) {
+            logger.info("Omega distribution {} is available for that engine build, downloading it", gameVersion.getOmegaNumber());
+            downloadURL = DownloadUtils.createFileDownloadUrlJenkins(gameVersion.getJob().getOmegaJobName(),
+                    gameVersion.getOmegaNumber(), DownloadUtils.FILE_TERASOLOGY_OMEGA_ZIP);
+        } else {
+            logger.warn("Engine build {} has no Omega zip available! Falling back to main zip without extra modules", buildNumber);
```

Comment:
This case is the scenario where an engine build in Jenkins has no associated Omega zip, which can happen in a few situations, It isn't fatal and there's a TODO elsewhere to mark it more clearly to the end user. Adding something new to the UI is a bit outside my reach at the moment though :-)

I was mainly thinking about newly downloaded games in which case you'd just lack modules, saves would work fine. But it is a valid point that if the user updates from an Omega release to an engine-only release then yeah saves would all kinds of crash. That needs to be in disclaimer somewhere. I'll split off an issue for it, thanks!